app/models/case_log.rb

```diff
@@ -212,6 +161,31 @@ def update_status!
                   else
                     "in_progress"
                   end
+    set_derived_fields
+  end
+
+  def set_derived_fields
+    self.postcode = UKPostcode.parse(property_postcode).outcode if property_postcode.present?
```

Comment:
Probably better to wrap these in a single `if` since it's the same condition for all 4?

Reply:
it's checking `property_postcode` for 2 of them and `previous_postcode` for the other 2, but I'll extract them anyway